homeassistant/components/cloud/client.py

```diff
@@ -136,6 +146,24 @@ async def get_google_config(self) -> google_config.CloudGoogleConfig:
 
         return self._google_config
 
+    def _cloud_assist_pipeline(self) -> str | None:
+        """Return the ID of a cloud-enabled assist pipeline or None."""
+        for pipeline in assist_pipeline.async_get_pipelines(self._hass):
+            if (
+                pipeline.conversation_engine == conversation.HOME_ASSISTANT_AGENT
+                and pipeline.stt_engine == DOMAIN
+                and pipeline.tts_engine == DOMAIN
+            ):
+                return pipeline.id
+        return None
+
+    async def _ensure_cloud_assist_pipeline(self) -> None:
+        """Ensure a cloud-enabled assist pipeline exists."""
+        if self._cloud_assist_pipeline():
+            return
+        await assist_pipeline.async_create_default_pipeline(self._hass)
+        self.cloud_pipeline = self._cloud_assist_pipeline()
```

Comment:
If the user delete the cloud pipeline or change tts and stt, will a new one be created at next restart ?

Reply:
Yeah, should we prevent that?

Reply:
Don't know. But It can be annoying if users don't want to use cloud but whisper and piper for stt/tts. They delete the cloud pipeline and it will came back at each restart.

Reply:
They could set the whisper/piper pipeline to preferred though, right?

Reply:
Yes but they can not delete the cloud one. It will be re-created at each restart.

Reply:
I agree it's dumb, let's fix it 👍